src/include/netloc_util.h

```diff
@@ -0,0 +1,47 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2001 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef NETLOC_UTIL_H_INCLUDED
+#define NETLOC_UTIL_H_INCLUDED
+
+#include "netloc.h"
+
+typedef enum {
+    MPIR_NETLOC_NETWORK_TYPE__FAT_TREE,
+    MPIR_NETLOC_NETWORK_TYPE__CLOS_NETWORK,
+    MPIR_NETLOC_NETWORK_TYPE__INVALID,
+} MPIR_Netloc_network_topo_type;
+
+typedef struct {
+    MPIR_Netloc_network_topo_type type;
+
+    union {
+        struct {
+            /* the levels at which the host and switch nodes are in
+             * the network */
+            int *node_levels;
+        } tree;
+    } u;
+} MPIR_Netloc_network_attributes;
+
+int MPIR_Netloc_parse_topology(netloc_topology_t topology,
+                               MPIR_Netloc_network_attributes * network_attr);
+
+int MPIR_Get_network_end_point(MPIR_Netloc_network_attributes, netloc_topology_t netloc_topology,
+                               hwloc_topology_t hwloc_topology, hwloc_cpuset_t bindset,
+                               netloc_node_t ** end_point);
```

Comment:
The function names are not consistent.  If it's related to netloc, can you use the `MPIR_Netloc_` prefix?

Reply:
This function is a utility I wrote to get netloc node for the current process. Does this name sound right? MPIR_Netloc_get_network_end_point